src/main/java/org/dasein/cloud/network/LbListener.java

```diff
@@ -102,6 +102,13 @@ static public LbListener getInstance(@Nonnull LbAlgorithm algorithm, @Nonnull Lb
         return new LbListener(algorithm, persistence, protocol, publicPort, privatePort);
     }
 
+    static public LbListener getInstance(@Nonnull LbAlgorithm algorithm, @Nonnull LbPersistence persistence, @Nullable String cookie, @Nonnull LbProtocol protocol, int publicPort, int privatePort, @Nullable String sslCertificateName, @Nullable String providerLBHealthCheckId) {
```

Comment:
Again, please don't add new signatures with some extra @Nullable fields. Add @Nullable fields with fluent interface methods `withXXX`. The purpose of `getInstance` method is not a shortcut to creating an instance in one go, but to construct a minimally complete object with only required fields being set. So unless the parameter is required, it shouldn't be set in `getInstance`.